sdc/tests/test_groupby.py

```diff
@@ -53,6 +54,272 @@
+                    'D': [np.nan, 2., -1.3, np.nan, 3.5, 0, 10, 0.42, np.nan, -2.5, 23],
+                    'E': ['a', 'bb', 'd', 'c', ' ', None, 'aaa', None, 'bb', None, 'd']
+        })
+        result = hpat_func(df)
+        result_ref = test_impl(df)
+        # TODO: implement index classes, as current indexes do not have names
+        pd.testing.assert_frame_equal(result, result_ref, check_names=False)
+
+    @skip_sdc_jit('Fails with old-pipeline from the start')
+    def test_dataframe_groupby_count_by_str(self):
+        def test_impl(df):
+            return df.groupby('A').count()
+        hpat_func = self.jit(test_impl)
+
+        n = 11
+        df = pd.DataFrame({
+                    'A': ['b', 'a', 'a', 'a', 'b', 'b', 'a', ' ', None, 'a', None],
+                    'B': np.arange(n, dtype=np.intp),
+                    'C': np.arange(n, dtype=np.float_),
+                    'D': [np.nan, 2., -1.3, np.nan, 3.5, 0, 10, 0.42, np.nan, -2.5, 23],
+                    'E': ['a', 'bb', 'd', 'c', ' ', None, 'aaa', None, 'bb', None, 'd']
+        })
+        result = hpat_func(df)
+        result_ref = test_impl(df)
+        # TODO: implement index classes, as current indexes do not have names
+        pd.testing.assert_frame_equal(result, result_ref, check_names=False)
+
+    @skip_sdc_jit('Fails with old-pipeline from the start')
+    def test_dataframe_groupby_sort(self):
+        def test_impl(df, param):
+            return df.groupby('A', sort=param).min()
+        hpat_func = self.jit(test_impl)
+
+        n = 11
+        df = pd.DataFrame({
+                    'A': ['b', 'a', 'a', 'a', 'b', 'b', 'a', ' ', None, 'a', None],
+                    'B': np.arange(n, dtype=np.intp),
+                    'C': np.arange(n, dtype=np.float_),
+                    'D': [np.nan, 2., -1.3, np.nan, 3.5, 0, 10, 0.42, np.nan, -2.5, 23]
+        })
+
+        for value in [True, False]:
+            with self.subTest(sort=value):
+                result = hpat_func(df, value) if value else hpat_func(df, value).sort_index()
+                result_ref = test_impl(df, value) if value else hpat_func(df, value).sort_index()
+                # TODO: implement index classes, as current indexes do not have names
+                pd.testing.assert_frame_equal(result, result_ref, check_names=False)
+
+    @skip_sdc_jit('Fails with old-pipeline from the start')
+    def test_dataframe_groupby_max(self):
+        def test_impl(df):
+            return df.groupby('A').max()
+        hpat_func = self.jit(test_impl)
+
+        n = 11
+        df = pd.DataFrame({
+                    'A': [2, 1, 2, 1, 2, 2, 1, 0, 3, 1, 3],
+                    'B': np.arange(n, dtype=np.intp),
+                    'C': np.arange(n, dtype=np.float_),
+                    'D': [np.nan, 2., -1.3, np.nan, 3.5, 0, 10, 0.42, np.nan, -2.5, 23],
+                    'E': [np.inf, 2., -1.3, -np.inf, 3.5, 0, 10, 0.42, np.nan, -2.5, 23]
```

Comment:
The data is used in a lot of places. Wouldn't you like to move it to some common place?

Reply:
Yes, you're probably right, I moved it to a global var.